test-data/unit/check-enum.test

```diff
@@ -1888,3 +1888,54 @@ def f2(c: C, a: Any) -> None:
     y = {'y': a, 'x': c.value}
     reveal_type(y)  # N: Revealed type is "builtins.dict[builtins.str*, Any]"
 [builtins fixtures/dict.pyi]
+
+[case testCanOverrideDunderAttributes]
+import typing
+from enum import Enum, Flag
+
+class BaseEnum(Enum):
+    __dunder__ = 1
+    __labels__: typing.Dict[int, str]
+
+class Override(BaseEnum):
+    __dunder__ = 2
+    __labels__ = {1: "1"}
+
+Override.__dunder__ = 3
+BaseEnum.__dunder__ = 3
+Override.__labels__ = {2: "2"}
+
+class FlagBase(Flag):
+    __dunder__ = 1
+    __labels__: typing.Dict[int, str]
+
+class FlagOverride(FlagBase):
+    __dunder__ = 2
+    __labels = {1: "1"}
+
+FlagOverride.__dunder__ = 3
+FlagBase.__dunder__ = 3
+FlagOverride.__labels__ = {2: "2"}
+[builtins fixtures/dict.pyi]
+
+[case testCanNotInitialize__members__]
+import typing
+from enum import Enum
+
+class WritingMembers(Enum):
+    __members__: typing.Dict[Enum, Enum] = {}  # E: Cannot declare read-only attribute "__members__"
```

Comment:
I think we can improve this error message. `__members__` is not read only, it more like "internal"?..

I would be quite lost with the current one.

Reply:
Yeah, agreed, I tried to generalise what's said in the documentation a bit: https://docs.python.org/3/library/enum.html#supported-dunder-names

> `__members__` is a read-only ordered mapping of `member_name:member` items. It is only available on the class.

Inlining it here just for reference

What do you think about `Cannot assign to internal attribute "__members__"`? Not completely sure if "assign" is most appropriate though

Reply:
Maybe exposing the internal implementation would be a good idea in this case?
Like `'Assigned "__members__" will be overriden by "Enum" internally'`?

Reply:
That reads good IMO, lets stick with that!